File: 1381-design-a-stack-with-increment-operation/1381-design-a-stack-with-increment-operation.java
class CustomStack {
    int[] stack;
    int idx;

    public CustomStack(int maxSize) {
        stack = new int[maxSize];
        idx = 0;

        for (int i = 0; i < maxSize; i++) {
            stack[i] = -1;
        }
    }
    
    public void push(int x) {
        if (idx < stack.length) {
            stack[idx] = x;
            idx++;
        }
    }
    
    public int pop() {
        if (stack[0] == -1) return -1;
        int result = stack[idx - 1];
        stack[idx - 1] = -1;
        idx--;
        return result;
    }
    
    public void increment(int k, int val) {
        for (int i = 0; i < Math.min(k, idx); i++) {
            stack[i] += val;
        }
    }
}

File: 1381-design-a-stack-with-increment-operation/1381-design-a-stack-with-increment-operation.py
class CustomStack:
    def __init__(self, maxSize: int):
        self.stack = [-1 for i in range(maxSize)]
        self.idx = 0

    def push(self, x: int) -> None:
        if self.idx < len(self.stack):
            self.stack[self.idx] = x
            self.idx += 1

    def pop(self) -> int:
        if self.stack[0] == -1: return -1
        result = self.stack[self.idx - 1]
        self.stack[self.idx - 1] = -1
        self.idx -= 1
        return result

    def increment(self, k: int, val: int) -> None:
        print(self.stack)
        for i in range(min(k, self.idx)):
            self.stack[i] += val
